services/batch/service/src/main/java/org/collectionspace/services/batch/nuxeo/botgarden/UpdatePlantPhenologyBatchJob.java

```diff
@@ -0,0 +1,135 @@
+package org.collectionspace.services.batch.nuxeo.botgarden;
+
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.HashMap;
+import java.util.ArrayList;
+
+import org.collectionspace.services.batch.nuxeo.AbstractBatchJob;
+import org.collectionspace.services.client.CollectionObjectClient;
+import org.collectionspace.services.common.NuxeoBasedResource;
+import org.collectionspace.services.common.invocable.InvocationContext;
+import org.collectionspace.services.common.invocable.InvocationResults;
+import org.collectionspace.services.common.invocable.InvocationContext.Params.Param;
+import org.collectionspace.services.common.ResourceMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This batch job updates the fruiting and flowering months, as well as the sex, fruit color and flower color of collectionobject records.
```

Comment:
Where are these fields defined? I know where `sex` is. I expected to see the rest in botgarden-collectionobject.xm, but I'm not seeing them: https://github.com/collectionspace/application/blob/master/tomcat-main/src/main/resources/tenants/botgarden/botgarden-collectionobject.xml

Contrary to this comment, this appears to allow updating _any_ field in the `collectionobjects_botgarden` schema (along with `sex` in `collectionobjects_common`). @remillet is that the intent? If so, the name of the batch job and the above comment should probably change to make it more clear.

Reply:
@remillet @rjaffe Also it would be good to know how this is used. Is there a UI? Or is this meant to be called by scripts?

Reply:
@ray-lee UCBG has a Update Plant Phenology data update tool in the UI. See attached screenshots. Happy to discuss further. - Rick

![ucbg qa collectionspace-Update-Plant-Phenology-data-update-tool-ui](https://github.com/collectionspace/services/assets/638932/680df04c-bbea-4975-accb-6fcc7f29aed7)
![ucbg qa collectionspace-data-update-tool-list](https://github.com/collectionspace/services/assets/638932/40d719d0-e98d-4106-b543-880d4a487fd5)

Reply:
Ok, I'm fine with this as long as we rename it to something descriptive of what it actually does, and change its comment to describe to what actually it does.